sstables/sstable_directory.hh

```diff
@@ -170,13 +171,13 @@ private:
 
     future<> process_descriptor(sstables::entry_descriptor desc, process_flags flags);
     void validate(sstables::shared_sstable sst, process_flags flags) const;
-    future<sstables::shared_sstable> load_sstable(sstables::entry_descriptor desc, sstables::sstable_open_config cfg = {}) const;
+    future<sstables::shared_sstable> load_sstable(sstables::entry_descriptor desc, sstables::sstable_open_config cfg = {}, bool skip_loading_data_and_index = false) const;
```

Comment:
Adding a new flag like this introduces a lot of churn (all callers have to be adjusted). We usually split this out into a separate commit, which just adds the flag and updates all callers, but it doesn't use it in the method yet.

Then we add the logic actually using the flag in the method in a 2nd commit. Lastly, a 3rd commit changes the interested callers to pass the non-default value of the flag. If all callers pass the same thing, than this can be folded into the 1st patch.

This amount of separation might seem superfluous, but it makes the review really easy, because reviewers can just glance over the chore patch (adding the flag) and go straight into reviewing the actually interesting changes.

Since this patch is not that big, I don't think you need to split it, just keep this in mind for future PRs.

Reply:
Sure. I'll keep that in mind for futures PRs.